### PR TITLE
⚠️ Remove v1a1 code

### PR DESCRIPTION
**Pull request by bryanv**
**What does this PR do, and why is it needed?**

This removes the bulk of the v1a1 code. Some v1a2 FFS checks are still there and will either be removed or kept as a guide for when we have to do this again. Similarly, there's also various FooA2() functions that I've kept as-is for now. Controllers and webhooks are also still under a v1a2 directory.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #

**Are there any special notes for your reviewer**:

There is some more package rename and clean up - like the provider vsphere2 - to do later. 

This still needs to go through internal builds, etc so I'm not going to get merged super soon.

**Please add a release note if necessary**:

```release-note
NONE
```

### CODE REVIEW

**Comment by akutz** [comment]
I'm concerned about the RBAC removals. For example, the RBAC for the old, v1a1 VM publish request. Won't our service account still require this? Or for that matter, the other v1a1 RBAC?

**Comment by bryanv** [comment]
> I'm concerned about the RBAC removals. For example, the RBAC for the old, v1a1 VM publish request. Won't our service account still require this? Or for that matter, the other v1a1 RBAC?

The PublishRequest RBAC should be and looks to be unchanged in this MR. The VMOP types RBAC removed here are for v1a1 only types with corresponding code that has not been enabled for some time since their FFS predatess v1a2, so this MR should remove the remaining k8s client references to them. The Namespace RBAC shouldn't be needed anymore since that info is now obtained from the AZ CRs but we can add that back if wanted since it will be required for some upcoming work.

File: pkg/prober/prober_manager.go [hunk 1]
```diff
@@ -60,24 +60,28 @@ type manager struct {
 	// if the time duration set in the AddAfter is not zero. vmReadinessProbeList can be used to avoid
 	// adding VMs to the readiness queue when this VM is already in the heap but not in the queue.
 	readinessMutex       sync.Mutex
-	vmReadinessProbeList map[string]*vmopv1.Probe
+	vmReadinessProbeList map[string]vmopv1.VirtualMachineReadinessProbeSpec
 }
 
 // NewManger initializes a prober manager.
-func NewManger(client client.Client, record vmoprecord.Recorder, vmProvider vmprovider.VirtualMachineProviderInterface) Manager {
+func NewManger(
```
Comment (sreyasn) [inline pkg/prober/prober_manager.go:67]: not your change, typo method name: `Manager`

**Review by sreyasn — APPROVED** [review]
Change looks good to me.

**Pull request merged**